src/lib/openapi/flatten-openapi-schema.test.ts

```diff
@@ -76,6 +113,65 @@ test('flattenOneOfSchema (object merging): merges properties and computes requir
   t.deepEqual(flattened.required, ['a'])
 })
 
+test('flattenOneOfSchema (object merging): merges enum values from properties', (t) => {
+  const schema: OpenapiSchema = {
+    oneOf: [
+      {
+        type: 'object',
+        properties: {
+          status: {
+            type: 'string',
+            enum: ['pending', 'processing'],
+          },
+          type: {
+            type: 'string',
+            enum: ['type1', 'type2'],
+          },
+        },
+      },
+      {
+        type: 'object',
+        properties: {
+          status: {
+            type: 'string',
+            enum: ['processing', 'completed'],
+          },
+          type: {
+            type: 'string',
+            enum: ['type2', 'type3'],
+          },
+        },
+      },
+    ],
+  }
+  const flattened = flattenOneOfSchema(schema as { oneOf: OpenapiSchema[] })
+
+  t.is(flattened.type, 'object')
+  t.truthy(flattened.properties)
+
+  if (flattened.properties != null) {
+    if (flattened.properties['status'] != null) {
+      const statusProp = flattened.properties['status'] as OpenapiSchema & {
+        enum: string[]
+      }
+      t.deepEqual(statusProp.enum, ['pending', 'processing', 'completed'])
+    } else {
+      t.fail('Expected "status" property to exist')
+    }
+
+    if (flattened.properties['type'] != null) {
+      const typeProp = flattened.properties['type'] as OpenapiSchema & {
+        enum: string[]
+      }
+      t.deepEqual(typeProp.enum, ['type1', 'type2', 'type3'])
+    } else {
+      t.fail('Expected "type" property to exist')
+    }
+  } else {
```

Comment:
This was a bit hard for me to follow. Should be simpler if you do all the nil checks up front.

```
if (foo == null) {
  t.fail('foo was null')
  return
}
```

Reply:
https://github.com/seamapi/blueprint/pull/166/commits/f02aab2f8c0a86d9d46e58360cf213a5af2f53eb